### PR TITLE
Adding Push/Pull functionality.

### PR DESCRIPTION
**Pull request by swernli**
@jterry75 @jstarks 
Adding new cmdlets Request/Submit-ContainerImage for Pull/Push functionality.

### CODE REVIEW

File: src/Docker.PowerShell/Cmdlets/RequestContainerImage.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+namespace Docker.PowerShell.Cmdlets
+{
+    [Cmdlet(VerbsLifecycle.Request, "ContainerImage",
+            DefaultParameterSetName = CommonParameterSetNames.Default)]
+    [OutputType(typeof(ContainerListResponse))]
+    [Alias("Pull-ContainerImage")]
+    public class RequestContainerImage : DkrCmdlet
+    {
+        private string _statusUpToDate = "Status: Image is up to date for ";
```
Comment (jterry75) [inline src/Docker.PowerShell/Cmdlets/RequestContainerImage.cs]: make these const

File: src/Docker.PowerShell/Cmdlets/RequestContainerImage.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            string repoTag = null;
+            bool failed = false;
+            var pullTask = DkrClient.Images.PullImageAsync(new ImagesPullParameters() { All = All, Parent = Repository, Tag = Tag, }, Authorization);
+            var messageWriter = new JsonMessageWriter(this);
+
+            using (var pullStream = await pullTask)
+            {
+                // ReadLineAsync is not cancellable without closing the whole stream, so register a callback to do just that.
+                using (CmdletCancellationToken.Register(() => pullStream.Close()))
+                using (var pullReader = new StreamReader(pullStream, new UTF8Encoding(false)))
```
Comment (jterry75) [inline src/Docker.PowerShell/Cmdlets/RequestContainerImage.cs]: I still wish this was done at the Docker.DotNet level using an `IProgress<T> where T : JsonMessage`  and then the pull task is just a task and its "stream" is sent as progress notifications. Makes 100% more sense to me than this model.

Comment (swernli) [inline src/Docker.PowerShell/Cmdlets/RequestContainerImage.cs]: I'll file an issue so we can follow up on this.

**Comment by swernli** [comment]
I think this is ready to merge if there are no other comments to address.

**Comment by swernli** [comment]
Added one last fix (if tag isn't specified, use "latest") to prevent downloading extra images.  Ready merge after signoff.

**Comment by jterry75** [comment]
LGTM

**Pull request merged**